AdyenCashAppPay/CashAppPayComponent.swift

```diff
@@ -0,0 +1,260 @@
+//
+// Copyright (c) 2023 Adyen N.V.
+//
+// This file is open source and available under the MIT license. See the LICENSE file for more info.
+//
+
+@_spi(AdyenInternal) import Adyen
+import PayKit
+import UIKit
+
+/// A component that handles a Cash App Pay payment.
+@available(iOS 13.0, *)
+public final class CashAppPayComponent: PaymentComponent,
+    PaymentAware,
+    PresentableComponent,
+    LoadingComponent {
+
+    /// The notification to post when returning back to your application from Cash App.
+    public static let RedirectNotification = CashAppPay.RedirectNotification
+
+    private enum ViewIdentifier {
+        static let storeDetailsItem = "storeDetailsItem"
+        static let cashAppButtonItem = "cashAppButtonItem"
+    }
+
+    /// The context object for this component.
+    @_spi(AdyenInternal)
+    public var context: AdyenContext
+
+    /// The payment method object for this component.
+    public var paymentMethod: PaymentMethod { cashAppPayPaymentMethod }
+
+    /// The delegate of the component.
+    public weak var delegate: PaymentComponentDelegate? {
+        didSet {
+            if let storePaymentMethodAware = delegate as? StorePaymentMethodFieldAware,
+               storePaymentMethodAware.isSession {
+                configuration.showsStorePaymentMethodField = storePaymentMethodAware.showStorePaymentMethodField ?? false
+            }
+        }
+    }
+
+    /// Component's configuration
+    public var configuration: CashAppPayConfiguration
+
+    public lazy var viewController: UIViewController = SecuredViewController(child: formViewController,
+                                                                             style: configuration.style)
+
+    public var requiresModalPresentation: Bool = true
+
+    private let cashAppPayPaymentMethod: CashAppPayPaymentMethod
+
+    private var storePayment: Bool {
+        configuration.showsStorePaymentMethodField ? storeDetailsItem.value : configuration.storePaymentMethod
+    }
+
+    private lazy var cashAppPay: CashAppPay = {
+        let cashAppPayKit = CashAppPay(clientID: cashAppPayPaymentMethod.clientId, endpoint: cashAppPayEnvironment)
+        cashAppPayKit.addObserver(self)
+        return cashAppPayKit
+    }()
+
+    private var cashAppPayEnvironment: CashAppPay.Endpoint {
+        guard let environment = context.apiContext.environment as? Environment else {
+            return .sandbox
+        }
+        return environment.isLive ? .production : .sandbox
```

Comment:
Just a thought:

with the same result, we can just check if `environment == .test` 🤔

Reply:
What about `local`, '`beta` etc?